### ISSUE TITLE
Missing scientific (domain-specific) images and corresponding examples and tutorials

### ISSUE THREAD

**Issue by emmanuelle** [scikit-image/scikit-image#3384]
I think scientific users could be more interested in starting using scikit-image if our documentation showed more examples corresponding to applications in different scientific domains. This could include cell images, satellite / remote sensing images, hyperspectral images, materials science images, astronomy images, etc. We could ask users to contribute some open data and examples, but before doing that is there a consensus that it would be useful?

**Comment by sciunto** [comment]
I think this is extremely valuable. However, we have the size issue... A discussion started here: #3323

**Comment by emmanuelle** [comment]
@sciunto There is the size issue indeed but it seems that it is due primarily to cython files; moreover the largest data files are the astronaut HOG data which are used only for testing, maybe it's possible to decrease their size by using a subimage.

**Comment by jni** [comment]
@emmanuelle just last week I gave a talk about scikit-image, and as I was browsing the gallery for useful examples, I noted this exact problem: we have a nice gallery of examples that feel too artificial to compel scientists. I ended up using the images from your blog posts. =) So, a big :+1: from me. We need to solve the scientific examples issue, and @sciunto if that means that we need to solve the size issue, then we need to solve that issue also. =)

Since we are considering moving data out of the core package and have it download on demand, it might be a good opportunity to revisit our example data also.

**Comment by emmanuelle** [comment]
We might use images from Wikipedia. I was quickly browsing through wikipedia microscopy images, 
https://en.wikipedia.org/wiki/Agranulocyte#/media/File:Monocyte_no_vacuoles.JPG
https://commons.wikimedia.org/wiki/File:Hairy_cell_leukemia_smear_2009-08-20.JPG
https://commons.wikimedia.org/wiki/File:Zz_Plagiomnium_undulatum_fluorescence.jpg
https://commons.wikimedia.org/wiki/File:NdFeB-Domains.jpg
Is CC-by-SA (needs attribution) license OK, or do we need public domain images?

**Comment by sciunto** [comment]
So far, we used only CC0 / public domain images, and If I'm correct, that was a requirement made by @stefanv I guess for the compatibility with the BSD license.

**Comment by stefanv** [comment]
Two issues arise: 1) how to communicate such requirements with users 
effectively and 2) do we want to place potentially surprising restrictions 
in place. For number 2, we've argued strongly "no": you should be able to 
use the output of scikit-image without breaching a patent or attribution 
requirement.

**Comment by hmaarrfk** [comment]
I would not use data size as an excuse not to include sample images. The
data issue is not something that will magically go away. Furthermore, I
think it is really important to run benchmarks on “out of cache” datasets.
Performance on small image sizes might be ok, but for images larger than a
few MB, it is unlikely that performance will ve the same since the data
wont fit in cache.

One way to get realistic images is to simply ask for them. We can ask for
them to be donated, maybe when people submit issues and we help them debug,
it can be appropriate to ask them to donate the image they posted under an
appropriate license. That would give us images that people actually care
about too!

**Comment by stefanv** [comment]
Actually, I do hope to make the data problem go away, by hosting the data externally, and downloading it as necessary.  We've been investigating tooling for that, such as https://github.com/frictionlessdata/datapackage-py, but currently it is too heavy a dependency.  However, the developers have indicated interest in improving that situation, and we can even write our own very simplistic downloader that handles their metadata spec.

**Comment by hmaarrfk** [comment]
I meant: "It won't magically go away if we don't add things to it" The package at 10s of MB is already too large. 

Cool package!

**Comment by emmanuelle** [comment]
Yes, CC0/public domain makes sense, thanks for your answers. Some packages using large datasets for their examples, like ``nilearn``, download them instead of shipping them (see for example http://nilearn.github.io/auto_examples/05_advanced/plot_neurovault_meta_analysis.html).

**Comment by GenevieveBuckley** [comment]
Yes to this, definitely. I'd also be ok with @stefanv's solution of not bundling large images, but instead allowing them to be downloaded on demand.

It's hard to get good CC0/public domain scientific images. There was an offer to donate a 3D dataset (https://github.com/scikit-image/scikit-image/issues/2513), that conversation should be revived.

I really want a 5D microscopy dataset available. If there is sufficient interest we (@jni and I) can ask someone at work if they can make us one and donate it, but we'd need to be clear about what we want it to be useful for.

**Comment by sciunto** [comment]
@GenevieveBuckley See also #2513

**Comment by stefanv** [comment]
A note that https://github.com/fatiando/pooch is being discussed [here on #3323](https://github.com/scikit-image/scikit-image/issues/3323)

**Comment by sciunto** [comment]
Do we start a new repository to push our new images?

**Comment by jni** [comment]
@sciunto I don't know whether a repository is the right place to store big images, compared to something like Figshare or the Open Science Framework. And if we are going to store *some* files on one of these storage platforms, perhaps we might as well store all of them?

**Comment by jni** [comment]
Just an fyi, MetPy is now using pooch, so that gives me a bit of extra confidence about using it.

https://github.com/Unidata/MetPy/pull/915

**Comment by braincharter** [comment]
Hi guys, if you need real 3D data for testing from the medical field involving Frangi-Ridge detections:
https://www.dropbox.com/sh/f7uutop5v09fecc/AAAK7QJsqAS9OJZnzgOe39-5a?dl=0

It contains a CLARITY (microscopy, large white vessels) sample, a susceptibility weighting image (black blood) and a time-of-flight (white blood). It is all in the nifti format (.nii).

**Comment by jni** [comment]
@braincharter what is the license on those files?

**Comment by emmanuelle** [comment]
For downloading images on request and only if they are not already cached, it might be interesting to take a look at what they nilearn team has written https://github.com/nilearn/nilearn/blob/master/nilearn/datasets/utils.py

**Comment by stefanv** [comment]
It's high on my priority list to check out [Pooch](https://www.fatiando.org/pooch/latest/), which does exactly that.  Hopefully soon!

**Comment by GenevieveBuckley** [comment]
I have some fluorescence microscopy data we could release under CC0 (I'm double checking this with our microscopists now).

They are:
* 2D RGB image of kidney tissue
* 2D image with four color channels, lily of the valley
* I also have a quasi-3D volume of the same kidney tissue (also with 3 color channels). There's only 16 slices in the z-stack because the sample was a very thin slice. So it might not be very interesting but you are welcome to it.

I took these while training on a Nikon inverted microscope, and the sample slides are nothing special (just the ones we get for free from manufacturers to test with).

**Comment by GenevieveBuckley** [comment]
I guess my question is:
1. If you want to take a look, how would you like me to share them with you?
2. What information do you need to know about the sample and imaging conditions?

**Comment by jni** [comment]
Awesome! They are going to be too big to stick in the repo directly, so if you share them online anywhere semi-useful (say, cloudstor), we can grab them and experiment with pooch (#3945). It would be super cool to use one or more of these in the tutorial at SciPy in 2 weeks.

As far as I know the long term public-facing storage problem isn't yet solved in general, but these will be relatively small, so Open Science Framework (osf.io) is probably a good option. I put the data from the skan paper there.

**Comment by emmanuelle** [comment]
@GenevieveBuckley  reading this conversation one year after :-). Would you be interested in adding these datasets to https://gitlab.com/scikit-image/data/ and write tutorials about these images?

**Comment by GenevieveBuckley** [comment]
I had forgotten, thanks for this! I might try and do it this weekend, for the sprint.

**Comment by GenevieveBuckley** [comment]
I've opened a PR over at scikit-image/data on GitLab

**Comment by sciunto** [comment]
@scikit-image/core Who has the rights on gitlab? This PR is pending: https://gitlab.com/scikit-image/data/-/merge_requests/6

**Comment by emmanuelle** [comment]
Thanks for the reminder @sciunto! For now it's Juan, Stéfan and me but we should add all the core devs who have an account on gitlab, so please ping me here or on zulip to give me your gitlab handle so that I can add you.

**Comment by emmanuelle** [comment]
@GenevieveBuckley I just commented in the gitlab PR (just one quick question). The images are gorgeous :-). Now we'll need tutorials using these beautiful images :-). Could you please explain what's to be seen in the images and what kind of image processing would typically be done on such images?

**Comment by grlee77** [comment]
@emmanuelle: I am also grlee77 on gitlab

**Comment by GenevieveBuckley** [comment]
Not so gorgeous, if you look at the clipping of the histogram at both the top and bottom of the histograms! I took them when I was learning to use the inverted Nikkon, so the purpose was not really about the quality of the images. But since I own those images outright I can donate them here, whereas most other stuff I work on requires other people's permission to share.

I'll put tutorials on my to-do list. I'm on leave next week, so I don't have a good estimate of when I might find time to start work on this.

**Comment by sciunto** [comment]
no pressure! thank you @GenevieveBuckley !

**Comment by emmanuelle** [comment]
Thanks @GenevieveBuckley ! Great to know that you have ideas of tutorials using these images. @mkcor could be interested in collaborating with you on these tutorials if you think it's a good idea (Marianne has been working on life-science-oriented tutorials thanks to the CZI grant). She is on vacation next week, so no hurry at all.

**Comment by mkcor** [comment]
> Thanks @GenevieveBuckley ! Great to know that you have ideas of tutorials using these images. @mkcor could be interested in collaborating with you on these tutorials if you think it's a good idea (Marianne has been working on life-science-oriented tutorials thanks to the CZI grant). She is on vacation next week, so no hurry at all.

Absolutely! Talk soon.
